Signal/src/ViewControllers/HomeView/HomeViewController.m

```diff
@@ -719,15 +699,6 @@ - (void)tableViewSetUp
     self.tableView.tableFooterView = [[UIView alloc] initWithFrame:CGRectZero];
 }
 
-- (BOOL)shouldShowMissingContactsPermissionView
-{
-    if (!self.contactsManager.systemContactsHaveBeenRequestedAtLeastOnce) {
```

Comment:
We introduced a guard such that this `systemContactsHaveBeenRequestedAtLeastOnce` was never being set if contacts access was denied. I think it's just simpler to check for the explicit denied state rather than "has been requested and not authorized"

Reply:
+1